### ISSUE TITLE
How to change the cifar to do binary classification？

### ISSUE THREAD

**Issue by anewlearner** [vlfeat/matconvnet#514]
The question was discussed in #47,and I tried both methods mentioned there,but I stil can't figure it out.
The first method.
1. Use lables of -1 and +1.
2.Change the layers in cifar as follows.
% 11 ip2
net.layers{end+1} = struct('type', 'conv', ...
                           'filters', 0.1*randn(1,1,64,2, 'single'),...% change from 10 to 2
                           'biases', zeros(1,2,'single'), ...% change from 10 to 2
                           'filtersLearningRate', 1, ...
                           'biasesLearningRate', 2, ...
                           'stride', 1, ...
                           'pad', 0) ;
3.Change the error type in cnn_train to 'binary'.
The result I got was as follows.
train: epoch 01:   1/131: 326.5 (326.5) Hz binerr: 120.240 objective: 0.171
train: epoch 01: 131/131: 377.3 (332.3) Hz binerr: 99.983 objective: 0.001
val: epoch 01:   1/389: 808.0 (808.0) Hz binerr: 100.000 objective: 0.000

The second method.
1.Use lables of 1 and 2.
2.Change the layers as above.
3.Change the error function in line 205, 
m = min(1, size(predictions,3)) ; %change from 5 to 1. So only the top1 error was calculated.
The result seems to be wrong.
train: epoch 01:   7/131: 341.0 (355.1) Hz objective: NaN top1err: 0.534 top5err: 0.534
val: epoch 01: 315/389: 865.8 (887.0) Hz objective: NaN top1err: 0.013 top5err: 0.013

Thanks!

**Comment by hwrdtm** [comment]
@anewlearner I am trying to achieve the same by using labels of +1 and -1, but my issue is that top1err and top5err are already at 0.000 after the first iteration in the first epoch! Moreover, the net seems to fit the training data so well but drastically terrible for testing data.

How did you resolve it finally? Thanks

**Comment by Addhi86** [comment]
@hwrdtm If you are doing Binary classification then change 'multiclass' to 'binary' in the CNN_TRAIN

**Comment by samo313** [comment]
I am doing binary classification by using  0 and 1. i deal with an error from softmaxloss.
would you please help me?
thanks.
